models/iosxr/ntp_global/iosxr_ntp_global.yaml

```diff
@@ -0,0 +1,225 @@
+---
+module: iosxr_ntp_global
+short_description: Manages ntp resource module
+description: This module configures and manages the attributes of  ntp on Cisco
+  IOSXR platforms.
+version_added: 2.5.0
+author: Ashwini Mhatre (@amhatre)
+notes:
+  - Tested against IOSXR 7.0.2.
+  - This module works with connection C(network_cli).
+options:
+  config:
+    description: A dictionary of ntp options
+    type: dict
+    suboptions:
+        access_group:
+          description: Control NTP access
+          type: list
+          elements: dict
+          suboptions:
+            ipv4: &ipv4
+              type: dict
+              description: Configure IPv4 access
+              suboptions:
+                peer: &peer
+                  type: str
+                  description: Provide full access
+                query_only: &query_only
+                  type: str
+                  description: Allow only control queries.
+                serve: &serve
+                  type: str
+                  description: Provide server and query access.
+                serve_only: &serve_only
+                  type: str
+                  description: Provide only server access.
+            ipv6: &ipv6
+              type: dict
+              description: Configure IPv6 access
+              suboptions:
+                peer: *peer
+                query_only: *query_only
+                serve: *serve
+                serve_only: *serve_only
+            vrf:
+              type: dict
+              description: Specify non-default VRF.
+              suboptions:
+                name:
+                  type: str
+                  description: Specify non-default VRF.
+                ipv4: *ipv4
+                ipv6: *ipv6
+        authenticate:
+          description: Authenticate time sources
+          type: bool
+        authentication_key:
+          description: Authentication key for trusted time sources
+          type: dict
+          suboptions:
+            key_number:
+              description: <1-65535>  Key number
+              type: int
+            clear:
```

Comment:
Can you please share the command for this? For NX-OS, this has been handled in this manner.

https://github.com/ansible-network/resource_module_models/pull/149/files#diff-f6fc3bc94c48edafdd7ee275b4dacb2553a99690c1edb5721a395df58691274aR67-R83

Reply:
RP/0/RP0/CPU0:ios(config)#ntp authentication-key 1 md5 encrypted test1
RP/0/RP0/CPU0:ios(config)#commit
Mon Aug 30 16:11:43.782 UTC
RP/0/RP0/CPU0:ios(config)#exit
RP/0/RP0/CPU0:ios#show running-config ntp
Mon Aug 30 16:11:49.784 UTC
ntp
 max-associations 10
 interface GigabitEthernet0/0/0/0
  multicast client 224.0.0.9
  multicast key 1
  multicast destination 224.0.0.8
 !
 authentication-key 1 md5 encrypted test1

Reply:
```
RP/0/RP0/CPU0:ios(config)#ntp authentication-key 1 md5 clear test1
RP/0/RP0/CPU0:ios(config)#commit
Mon Aug 30 16:10:51.694 UTC
RP/0/RP0/CPU0:ios(config)#exit
RP/0/RP0/CPU0:ios#show running-config ntp
Mon Aug 30 16:10:59.886 UTC
ntp
 max-associations 10
 interface GigabitEthernet0/0/0/0
  multicast client 224.0.0.9
  multicast key 1
  multicast destination 224.0.0.8
 !
 authentication-key 1 md5 encrypted 00101615100A
```

Reply:
i think in this case I need to update clear option with no_log right?

Reply:
`authentication_keys` should be a list afaik. There can be multiple keys identified by the `id`.
For consistency, can we have the auth_key stored in `key`? Since there can only be 2 types of keys (clear-text and encrypted), can we make `encrypted` a bool?

```yaml
authentication_key:
  - id: 1
    key: test1
    encrypted: True
  - id: 2
    key: test2
    encrypted: False
```

Reply:
Note: Adding `no_log: True` will trigger validate-modules since `no_log` is not a part of doc schema (yet). You will have to remove this from the docstring once scaffolding is done.